polyfill/lib/duration.mjs

```diff
@@ -377,6 +377,120 @@ export class Duration {
     if (!ES.IsTemporalDuration(result)) throw new TypeError('invalid result');
     return result;
   }
+  round(options) {
+    if (!ES.IsTemporalDuration(this)) throw new TypeError('invalid receiver');
+    let years = GetSlot(this, YEARS);
+    let months = GetSlot(this, MONTHS);
+    let weeks = GetSlot(this, WEEKS);
+    let days = GetSlot(this, DAYS);
+    let hours = GetSlot(this, HOURS);
+    let minutes = GetSlot(this, MINUTES);
+    let seconds = GetSlot(this, SECONDS);
+    let milliseconds = GetSlot(this, MILLISECONDS);
+    let microseconds = GetSlot(this, MICROSECONDS);
+    let nanoseconds = GetSlot(this, NANOSECONDS);
+
+    let defaultLargestUnit = 'nanoseconds';
+    for (const [prop, v] of Object.entries({
+      years,
+      months,
+      weeks,
+      days,
+      hours,
+      minutes,
+      seconds,
+      milliseconds,
+      microseconds
+    })) {
+      if (v !== 0) {
+        defaultLargestUnit = prop;
+        break;
+      }
+    }
+
+    options = ES.NormalizeOptionsObject(options);
+    const smallestUnit = ES.ToSmallestTemporalDurationUnit(options, 'nanoseconds');
+    const relativeTo = ES.ToRelativeTemporalObject(options);
+    let largestUnit = ES.ToLargestTemporalUnit(options, defaultLargestUnit);
```

Comment:
Are you sure this needs to be `let`?

Reply:
Well spotted, left over from an earlier draft.